### ISSUE TITLE
standalone: conflicting types for 'strerror_r'

### ISSUE THREAD

**Issue by OmeGak** [Linuxbrew/legacy-linuxbrew#662]
I just installed Linuxbrew standalone on a clean Elementary OS Freya (Ubuntu 14.04). I'm again facing problems when installing Postgres but the issue seems to be related more to the standalone installation.

This is the output I get from brew:

```
brew install postgres               
==> Downloading https://ftp.postgresql.org/pub/source/v9.4.5/postgresql-9.4.5.tar.bz2
Already downloaded: /home/omegak/.cache/Homebrew/postgresql-9.4.5.tar.bz2
==> ./configure --prefix=/home/omegak/.linuxbrew/Cellar/postgresql/9.4.5 --datadir=/home/omegak/.linuxbrew/Cellar/postgresql/9.4.5/share/postgresql --docdir=/home/omegak/.linuxbr
Last 15 lines from /home/omegak/.cache/Homebrew/Logs/postgresql/01.configure:
checking whether pthreads work with -mt... no
checking for the pthreads library -lpthreadGC2... no
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for strerror_r... yes
checking for getpwuid_r... yes
checking for gethostbyname_r... yes
checking whether getpwuid_r takes a fifth argument... yes
checking whether strerror_r returns int... no
checking test program... failed
configure: error: 
Could not execute a simple test program.  This may be a problem
related to locating shared libraries.  Check the file 'config.log'
for the exact reason.
```

And this is the exact reason I found in `config.log`:

```
configure:12923: checking whether strerror_r returns int
configure:12944: /home/omegak/.linuxbrew/bin/gcc-5 -c -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Os -w -pipe -march=core2  -pthread  -D_REENTRANT -D_THREAD_SAFE -D_POSIX_PTHREAD_SEMANTICS -isystem/home/omegak/.linuxbrew/include -D_GNU_SOURCE -I/home/omegak/.linuxbrew/Cellar/libxml2/2.9.2/include/libxml2  conftest.c >&5
conftest.c: In function 'main':
conftest.c:158:5: error: conflicting types for 'strerror_r'
 int strerror_r(int, char *, size_t);
     ^
In file included from conftest.c:153:0:
/home/omegak/.linuxbrew/include/string.h:438:14: note: previous declaration of 'strerror_r' was here
 extern char *strerror_r (int __errnum, char *__buf, size_t __buflen)
              ^
```

What I can gather is that `libxml2` is redefining `strerror_r` to return `int` while the header in `string.h` specifies defines it to return `char*`. I have googled for a while but I don't see anybody else facing this problem and I have no clear idea on how to tackle it. Could anybody with more experience with a standalone installation be able to lend me a hand here?

This is the full log:
https://gist.github.com/anonymous/edb2b9d6e46b4f21943a

**Comment by OmeGak** [comment]
A bit more info. `include/string.h` is pointing to `Cellar/glibc/2.19/include/string.h`. 

I think I've found the source of discrepancy: http://man7.org/linux/man-pages/man3/strerror.3.html

```c
int strerror_r(int errnum, char *buf, size_t buflen);
    /* XSI-compliant */

char *strerror_r(int errnum, char *buf, size_t buflen);
    /* GNU-specific */
```

For some reason, the glibc version installed through homebrew comes with the GNU-specific specification of `strerror_r`, while `libxml2` expects the XSI-compliant.

**Comment by OmeGak** [comment]
Dirty hack saved the day. I managed to workaround the problem by editing `string.h` and returning `int` as expected by `libxml2`. I'd still like to find the reason why this happened and, if possible, help finding a clean solution.

I also found another caveat with `ncurses` along the way to install Posgres. This time, `libxml2` was trying to use `libncursesw.so.5`, but brew installs `libncursesw.so.6`. I simply symlinked the missing library to the one installed and the installation went fine from then on. I don't know if it's related to the initial issue.
